Description: Fix linjansson & libcjose detection
 Those two libs were wrongly detected on Debian. Now, we check for dynamic lib
 and only fall back on static if first fails.
Author: Jean Baptiste Favre <debian@jbfavre.org>
Origin: other
Bug: https://github.com/apache/trafficserver/pull/4296
Forwarded: https://github.com/apache/trafficserver/pull/4296
Applied-Upstream: 8.0.1
Last-Update: 2018-09-24
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: trafficserver/configure.ac
===================================================================
--- trafficserver.orig/configure.ac	2018-09-26 19:36:04.768402746 +0200
+++ trafficserver/configure.ac	2018-09-26 19:36:04.764402715 +0200
@@ -1306,21 +1306,21 @@
 #
 AC_CHECK_HEADERS([jansson.h], [
   AC_MSG_CHECKING([whether jansson is dynamic])
-  TS_LINK_WITH_FLAGS_IFELSE([-shared -fPIC -l:libjansson.a],[AC_LANG_PROGRAM(
+  TS_LINK_WITH_FLAGS_IFELSE([-fPIC -ljansson],[AC_LANG_PROGRAM(
                             [#include <jansson.h>],
                             [(void) json_object();])],
-                            [AC_MSG_RESULT([no]);  LIBJANSSON=-l:libjansson.a],
-                            [AC_MSG_RESULT([yes]); LIBJANSSON=-ljansson])
+                            [AC_MSG_RESULT([yes]); LIBJANSSON=-ljansson],
+                            [AC_MSG_RESULT([no]);  LIBJANSSON=-l:libjansson.a])
   ],
   [LIBJANSSON=])
 
 AC_CHECK_HEADERS([cjose/cjose.h], [
   AC_MSG_CHECKING([whether cjose is dynamic])
-  TS_LINK_WITH_FLAGS_IFELSE([-shared -fPIC -l:libcjose.a],[AC_LANG_PROGRAM(
+  TS_LINK_WITH_FLAGS_IFELSE([-fPIC -lcjose],[AC_LANG_PROGRAM(
                             [#include <cjose/cjose.h>],
                             [(void) cjose_jws_import("", 0, NULL);])],
-                            [AC_MSG_RESULT([no]);  LIBCJOSE=-l:libcjose.a],
-                            [AC_MSG_RESULT([yes]); LIBCJOSE=-lcjose])
+                            [AC_MSG_RESULT([yes]); LIBCJOSE=-lcjose],
+                            [AC_MSG_RESULT([no]);  LIBCJOSE=-l:libcjose.a])
   ],
   [LIBCJOSE=])
 AC_CHECK_LIB([crypto],[HMAC],[has_libcrypto=1],[has_libcrypto=0])
